lib/api.es6.js

```diff
@@ -50,6 +47,102 @@ const login = (apiOptions, username, pass) => {
   });
 };
 
+const register = (apiOptions, username, password, email, newsletter) => {
+  return new Promise((r, x) => {
+    if (!apiOptions.oauthAppOrigin) {
+      x('Please set up a Reddit Oauth App, and pass in its URL as oauthAppOrigin to config.');
+    }
+
+    if (!apiOptions.clientId) {
+      x('Please set up a Reddit Oauth App, and pass in its id as clientId to config.');
+    }
+
+    if (!apiOptions.clientSecret) {
+      x('Please set up a Reddit Oauth App, and pass in its secret as clientSecret to config.');
+    }
+
+    const data = {
+      user: username,
+      passwd: password,
+      passwd2: password,
+      api_type: 'json',
+    };
+
+    if (email) {
+      data.email = email;
+    }
+
+    if (newsletter) {
+      data.newsletter_subscribe = true;
+
+      if (!email) {
+        x('NEWSLETTER_NO_EMAIL');
```

Comment:
I was caught in between several conflicting protocols for error messages that I didn't put much thinking time into resolving. 

I was using the error strings from the login function but then I found out that errors strings could also come from making calls to `r2`, which are like the "token" type of messages. Morever, these errors strings aren't properly handled on the frontend. For login, any type of error gives an `Sorry that's not the right password` error, whereas I wanted more granular and helpful error messages for register. 

Straddling between consistency versus helpfulness, I hot-patched this frankenstein together.

I think what would really help is if I knew where I would consolidate error messages, then they won't be thrown everywhere